src/core/server/saved_objects/migrations/actions/check_for_unknown_docs.ts

```diff
@@ -39,33 +40,39 @@ export interface UnknownDocsFound {
 export const checkForUnknownDocs =
   ({
     client,
+    ignoreUnknownObjects,
     indexName,
     unusedTypesQuery,
     knownTypes,
   }: CheckForUnknownDocsParams): TaskEither.TaskEither<
     RetryableEsClientError | UnknownDocsFound,
-    {}
+    UnknownDocsFound | {}
   > =>
   () => {
     const query = createUnknownDocQuery(unusedTypesQuery, knownTypes);
-
     return client
       .search<SavedObjectsRawDocSource>({
         index: indexName,
         body: {
+          size: 1000,
           query,
         },
       })
       .then((body) => {
         const { hits } = body.hits;
-        if (hits.length) {
-          return Either.left({
-            type: 'unknown_docs_found' as const,
-            unknownDocs: hits.map((hit) => ({ id: hit._id, type: hit._source?.type ?? 'unknown' })),
-          });
-        } else {
+        if (!hits.length) {
           return Either.right({});
         }
+
+        const response = {
+          type: 'unknown_docs_found' as const,
+          unknownDocs: hits.map((hit) => ({
+            id: hit._id,
+            type: hit._source?.type ?? 'unknown',
+          })),
+        };
+
+        return ignoreUnknownObjects ? Either.right(response) : Either.left(response);
```

Comment:
I wonder if that's the correct way to do it. I would have returned `Either.right` either way, and have the model handle the logic of following by an error or not. @rudolf wdyt?

Reply:
I thought about doing it on `model.ts` as well, but I put it here mainly for 2 reasons:
- `model.ts` is already huge and has a lot of logic.
- From what I understood, the semantics of _Left_ and _Right_ seem to indicate whether the thing went _Wrong_ or it went _Right_. If a user decides to ignore uknown documents, then it kind of made sense to say: - We found some unknown objects, but that's al**right**, we can proceed with the migration.

It's not set in stone though, so if y'all feel more comfortable with having it on `model.ts` I can update it.

Reply:
Yea, I'm not sure either. Let's see what @rudolf thinks about it

Reply:
I agree with Left being failure and right success. I guess the question is how much should the action decide and this is a rather fuzzy thing that's probably mostly personal preference... 

I've usually tried to keep actions limited to the Elasticsearch domain, that is they understand how Elasticsearch works, but they don't understand how migrations work. The model ideally shouldn't have to worry too much about how Elasticsearch works, but it makes all the migration-related decisions. 

As an example when cloning an index, if the index already exists the action waits for the existing index to become yellow. The model doesn't need to know those details it just says to the action "tell me if you succeed or failed in cloning". (There's probably counter examples where we didn't keep this kind of separation too).

So I kinda agree with @pgayvallet that actions don't need to know about the `ignoreUnknownObjects` flag. The action could just find 0 or more unknown objects as a right (and if there's some exception trying to find them, it would return a left to say "I couldn't determine how many unknown objects there are").

There's a lot of logic in the model and it's quite complex, but one advantage is that it's fairly simple to test without needing any mocks. So I think the testability makes it more advantageous to have logic in the model instead of the action.